src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultStr = '';
+
+    if (this.length !== 0) {
```

Comment:
4. [CODE STYLE] - avoid using nested `if` statements, we are sure that you can do it without them :)
you can do: 
if (this.length !== 0) {
  return resultStr;
}